util/settings/settings.go

```diff
@@ -837,21 +861,64 @@ func (a *ArgoCDSettings) GetAppInstanceLabelKey() string {
 	return a.AppInstanceLabelKey
 }
 
-func (a *ArgoCDSettings) getExcludedResources() []ExcludedResource {
-	coreExcludedResources := []ExcludedResource{
+func (a *ArgoCDSettings) getExcludedResources() []FilteredResource {
+	coreExcludedResources := []FilteredResource{
 		{APIGroups: []string{"events.k8s.io", "metrics.k8s.io"}},
 		{APIGroups: []string{""}, Kinds: []string{"Event"}},
 	}
 	return append(coreExcludedResources, a.ResourceExclusions...)
 }
 
-func (a *ArgoCDSettings) IsExcludedResource(apiGroup, kind, cluster string) bool {
+func (a *ArgoCDSettings) checkResourcePresence(apiGroup, kind, cluster string, filteredResources []FilteredResource) bool {
 
-	for _, excludedResource := range a.getExcludedResources() {
-		if excludedResource.Match(apiGroup, kind, cluster) {
+	for _, includedResource := range filteredResources {
+		if includedResource.Match(apiGroup, kind, cluster) {
 			return true
 		}
 	}
 
 	return false
 }
+
+func (a *ArgoCDSettings) isIncludedResource(apiGroup, kind, cluster string) bool {
+	return a.checkResourcePresence(apiGroup, kind, cluster, a.ResourceInclusions)
+}
+
+func (a *ArgoCDSettings) isExcludedResource(apiGroup, kind, cluster string) bool {
+	return a.checkResourcePresence(apiGroup, kind, cluster, a.getExcludedResources())
+}
+
+// Behavior of this function is as follows:
```

Comment:
I think you're describing test cases :). Could you write a [table driven test](https://github.com/golang/go/wiki/TableDrivenTests)?